GLAA.Web/FormLogic/LicenceApplicationFormDefinition.cs

```diff
@@ -38,11 +52,115 @@ public bool CanViewNextModel<TParent>(FormSection section, int id, TParent paren
             return model == null || model.CanView(parent);
         }
 
+        public bool CanViewPage<TParent>(FormSection section, string actionName, TParent parent)
+        {
+            var page = GetPageDefinition(section, actionName);
+
+            if (page.OverrideViewCondition)
+            {
+                return true;
+            }
+
+            var model = page.GetViewModelExpressionForPage(parent) as ICanView<TParent>;
+
+            return model == null || model.CanView(parent);
+        }
+
         public int GetSectionLength(FormSection section)
         {
-            return fieldConfiguration.Fields[section].Length;
+            return fieldConfiguration[section].Length;
         }
 
+        public FormPageDefinition GetLastPage(FormSection section)
+        {
+            return fieldConfiguration[section].Last();
+        }
+
+        public bool IsNextPageLastPage(FormSection section, string actionName)
+        {
+            return fieldConfiguration[section][GetSectionLength(section) - 2].MatchesName(actionName);
```

Comment:
What's `-2` here? could we name it as a variable. Don't mean to be picky but seen so much code over the years using magic numbers which becomes hard to reason about later on (like people just going, oh I need it one more than this in my specific situation, I'll make it -1) and then the world ends!

Reply:
Fair point. The item at index `length - 2` is the penultimate item in the list, where the final item is usually the summary screen. I'll add some comments to this method to explain what's going on as it's not immediately obvious how it works (and the name doesn't help!)